### ISSUE TITLE
About the "ignore_mask" in the training code

### ISSUE THREAD

**Issue by yasuobinggan** [LiheYoung/UniMatch#22]
Amazing work! simple and effective!

**I noticed that the build of "ignore_mask" requires the "mask"(dataset/semi.py)**
![image](https://user-images.githubusercontent.com/30293157/228421039-531cee57-e58f-48a1-8946-175461305b8c.png)

**Then, I noticed that "ignore_mask", "ignore_mask_mix" are involved in building the loss, so is there some label(gt mask) information used in the calculation of unlabeled data loss?(unimatch.py, also in fixmatch.py)**
![image](https://user-images.githubusercontent.com/30293157/228421447-a72e18aa-663c-4d56-9da2-558f667de637.png)
![image](https://user-images.githubusercontent.com/30293157/228421461-a04fc2b2-acee-43f5-adfb-02fce81045f2.png)
![image](https://user-images.githubusercontent.com/30293157/228421487-c909935c-d2ea-4518-b9e0-a5fb20631959.png)

Thank you!

**Comment by LiheYoung** [comment]
Hi, thank you for raising this question.

Actually, ``ignore_mask`` or ``ignore_mask_mix`` does NOT contain any groundtruth information. It only marks the **padded region** during image pre-processing (resizing and cropping). As for the line ``ignore_mask[mask == 254] = 255``, the value ``254`` in ``mask`` is only labeled by the [padding practice](https://github.com/LiheYoung/UniMatch/blob/3ab072d4a1d9e7b76949ad158bc87c40713b70fd/dataset/semi.py#L42-L43). We use value ``254`` to distinguish the padded region from the manual defined ignored region (value ``255`` in Pascal).

Therefore, we do NOT use the semantic information in ``mask``, and do NOT use the ignored region (value ``255``) defined by manual anotators in ``mask``.

**Comment by yasuobinggan** [comment]
Get it! Thanks for your reply!

**Comment by yurithefury** [comment]
Just a follow-up question, how come you define `ignore_mask` in SemiCDDataset but not in ACDCDataset?
